src/model/Enemy.hs

```diff
@@ -2,130 +2,112 @@
-enemyIsHit :: Enemy e => e -> Bullet -> Maybe (e, Bullet)
-enemyIsHit e b | pos e == bulletPos b = Just (e, b) -- Enemy is hit. Change to this later: if the position is approximately equal.
-               | otherwise = Nothing -- Enemy is not hit.
+  -- Given an enemy and a bullet, return these arguments if the enemy is hit for despawning both. 
+  hit :: a -> Bullet -> Maybe (a, Bullet)
+  hit e b | pos e == bulletPos b = Just (e, b) -- Enemy is hit. Change to this later: if the position is approximately equal.
+          | otherwise = Nothing -- Enemy is not hit.
+  -- Render enemy at a random vertical position.
+  spawn :: a
+  -- Return the position of an Enemy
+  pos :: a -> Position
 
 -- Enemy types as instances of type class Enemy (Too much repitition?)
 instance Enemy BasicEnemy where
-        hit = enemyIsHit
-        spawn = BasicEnemy {basicEnemyPos = (screenWidth, randomY) , basicEnemyCooldown = cooldown} -- Spawn an enemy at right with a random y pos.
-                where -- Default value, change this later.
-                        cooldown = 2
-        pos = basicEnemyPos
+  spawn = BasicEnemy {basicEnemyPos = (screenMax, randomY) , basicEnemyCooldown = enemyShootingCooldown} -- Spawn an enemy at right with a random y pos.
+  pos = basicEnemyPos
 
 instance Enemy BurstEnemy where
-        hit = enemyIsHit
-        spawn = BurstEnemy {burstEnemyPos = (screenWidth, randomY) , burstEnemyCooldown = cooldown}  
-                where
-                        cooldown = 3
-        pos = burstEnemyPos
+  spawn = BurstEnemy {burstEnemyPos = (screenMax, randomY) , burstEnemyCooldown = enemyShootingCooldown}  
+  pos = burstEnemyPos
 
 instance Enemy ConeEnemy where
-        hit = enemyIsHit
-        spawn = ConeEnemy {coneEnemyPos = (screenWidth, randomY) , coneEnemyCooldown = cooldown}    
-                where
-                        cooldown = 2
-        pos = coneEnemyPos
+  spawn = ConeEnemy {coneEnemyPos = (screenMax, randomY) , coneEnemyCooldown = enemyShootingCooldown}    
+  pos = coneEnemyPos
 
 instance Enemy BasicPlayerSeekingEnemy where
-        hit = enemyIsHit
-        spawn = BasicPlayerSeekingEnemy {basicSeekingPos = (screenWidth, randomY) , basicSeekingCooldown = cooldown}      
-                where
-                        cooldown = 1
-        pos = basicSeekingPos
+  spawn = BasicPlayerSeekingEnemy {basicSeekingPos = (screenMax, randomY) , basicSeekingCooldown = enemyShootingCooldown}      
+  pos = basicSeekingPos
 
 instance Enemy FastPlayerSeekingEnemy where
-        hit = enemyIsHit
-        spawn = FastPlayerSeekingEnemy {fastSeekingPos = (screenWidth, randomY)}     
-        pos = fastSeekingPos
+  spawn = FastPlayerSeekingEnemy {fastSeekingPos = (screenMax, randomY)}     
+  pos = fastSeekingPos
        
 
 -- All enemies, except the player seeking types, are instances of this class.
-class MoveBasic a where 
-        moveBasic :: a -> a 
+class Enemy a => MoveBasic a where 
```

Comment:
Dit zorgt ervoor dat ik gebruik kan maken van `pos` in de `moveBasic` implementaties eronder.